### PR TITLE
Configure contribution status for invoices that can be pushed to xero

### PR DESCRIPTION
**Pull request by jitendrapurohit**
Contribution status not enabled on Xero Settings page would not be processed for invoice push.

### CODE REVIEW

File: CRM/Civixero/Invoice.php [hunk 1]
```diff
@@ -139,6 +139,15 @@ public function push($params, $limit = 10) {
       $errors = array();
 
       foreach ($records['values'] as $record) {
+        $status = civicrm_api3('Contribution', 'getvalue', array(
```
Comment (eileenmcnaughton) [inline CRM/Civixero/Invoice.php]: Can we move this to getContributionsRequiringPushUpdate function?

File: CRM/Civixero/Form/XeroSettings.php [hunk 2]
```diff
@@ -19,6 +19,9 @@ function buildQuickForm() {
         if ($add == 'addElement') {
           $this->$add($setting['html_type'], $name, ts($setting['title']), CRM_Utils_Array::value('html_attributes', $setting, array ()));
         }
+        elseif ($setting['html_type'] == 'Select') {
```
Comment (eileenmcnaughton) [inline CRM/Civixero/Form/XeroSettings.php]: makes sense

File: settings/Civixero.setting.php [hunk 3]
```diff
@@ -104,4 +104,23 @@
...
+    'name' => 'xero_push_contribution_status',
+    'type' => 'Array',
+    'add' => '4.7',
+    'is_domain' => 1,
+    'is_contact' => 0,
+    'description' => 'Select contribution status that can be pushed to Xero.',
+    'title' =>  'Push Contribution Status',
+    'help_text' => '',
+    'html_type' => 'Select',
+    'options' => CRM_Contribute_PseudoConstant::contributionStatus(NULL, 'name'),
```
Comment (eileenmcnaughton) [inline settings/Civixero.setting.php]: I think this syntax is 'more correct' https://github.com/litespeedmarc/civicrm-core/blob/dfc3d2e798940e3837e842e41e3d16a009760764/settings/Core.setting.php#L46-L46

**Comment by eileenmcnaughton** [comment]
Looks good - stylistic feedback

**Comment by jitendrapurohit** [comment]
Have updated the PR with the fix mentioned in the comments. Thanks.

File: CRM/Civixero/Invoice.php [hunk 4]
```diff
@@ -389,6 +380,18 @@ protected function getContributionsRequiringPushUpdate($params, $limit) {
     }
 
     $records = civicrm_api3('AccountInvoice', 'get', $criteria);
+
+    $pushEnabledStatuses = Civi::settings()->get('xero_push_contribution_status');
```
Comment (eileenmcnaughton) [inline CRM/Civixero/Invoice.php]: Actually - I was imagining combining the statues with this line

https://github.com/jitendrapurohit/nz.co.fuzion.civixero/blob/2f719b97a1cb14f5cbd36628c3032b25de69ac78/CRM/Civixero/Invoice.php#L371

We should probably have some defaults (Completed, Pending) & then not have the isNotCancelled thingee

**Pull request merged**